File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-4.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        char[] chs = s.toCharArray();
        int res = 0;
        int n = chs.length;
        int left = 0;
        int[] count = new int[128];
        for (int right = 0; right < n; right++) {
            char c = chs[right];
            count[c]++;
            while (count[c] > 1) {
                count[chs[left]]--;
                left++;
            }
            res = Math.max(right - left + 1, res);
        }
        return res;
    }
}

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-4.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        count = defaultdict(int)
        count[0] = 1
        res = s = 0
        for x in nums:
            s += x
            res += count[s - k]
            count[s] += 1
        return res